### DISCUSSION TITLE
Support for Unit and Serving Size Conversions

### DISCUSSION THREAD

**Discussion by RPGillespie6** [cooklang/spec#142] [General]
Seems like one of the big benefits of cooklang is that you could programmatically adjust a recipe for a different serving size, convert imperial to metric units, or vice versa. 

Any plans for this? Are PRs welcome? Should I just write my own tool?

**Comment by fellmann** [comment]
I was really surprised this is not implemented - seems like the most important feature and huge benefit!

**Comment by SkepticMystic** [comment]
Huh, I also thought this was implemented already. +1 for this FR

**Comment by stevenbell** [comment]
Scaling a recipe is supported with `cook recipe --scale SCALE_FACTOR`.

Unit conversion is not supported as far as I can tell.  It would be awesome, both for scaling up and down (8x a recipe with 1 tablespoon should give 1/2 cup instead of 8 T; or 2x 500g should give 1kg instead of 1000g) and for converting between imperial/metric.

My current use case is that I'll be cooking for ~100 people for a week, and being able to quickly scale recipes by large quantities would be super helpful.  (And yes, this would be trivial if I were doing everything by weight in the metric system... but sadly I'm not!)
